vdp/pipeline/v1beta/pipeline.proto

```diff
@@ -44,34 +44,149 @@ message ReadinessResponse {
+    // Instill UI order.
+    int32 instill_ui_order = 4 [(google.api.field_behavior) = OPTIONAL];
+    // Instill UI Multiline.
+    int32 instill_ui_multiline = 5 [(google.api.field_behavior) = OPTIONAL];
+  }
+  // Fields configuration.
+  // Key: Key of the input data.
+  // Field: Field settings of the value.
+  map<string, Field> fields = 1 [(google.api.field_behavior) = OPTIONAL];
+}
+
+// EndComponent
+// Configures the ending point for pipeline triggering.
+message EndComponent {
+  // Represents a field within the end component.
+  message Field {
+    // Title of the field.
+    string title = 1 [(google.api.field_behavior) = OPTIONAL];
+    // Description of the field.
+    string description = 2 [(google.api.field_behavior) = OPTIONAL];
+    // Value of the field.
+    string value = 3 [(google.api.field_behavior) = OPTIONAL];
+    // Instill UI order.
+    int32 instill_ui_order = 4 [(google.api.field_behavior) = OPTIONAL];
+  }
+  // Fields configuration.
+  // Key: Key of the output data.
+  // Field: Field settings of the value.
+  map<string, Field> fields = 1 [(google.api.field_behavior) = OPTIONAL];
+}
+
+// ConnectorComponent
+// Configures a connector component. Requires the creation of a connector resource first.
+message ConnectorComponent {
+  // Definition name.
+  string definition_name = 1 [(google.api.field_behavior) = OPTIONAL];
+  // Connector definition.
+  ConnectorDefinition definition = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Connector name. Leave empty if not created.
+  string connector_name = 3 [(google.api.field_behavior) = OPTIONAL];
+  // Describes the connector details (e.g., configuration to connect with the vendor service).
+  Connector connector = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Task.
+  string task = 5 [(google.api.field_behavior) = OPTIONAL];
+  // Input configuration of the component. JSON schema described in the connector definition.
+  google.protobuf.Struct input = 6 [(google.api.field_behavior) = OPTIONAL];
+  // Condition statement determining whether the component is executed or not.
+  string condition = 7 [(google.api.field_behavior) = OPTIONAL];
+}
+
+// OperatorComponent
+// Configures an operator component.
+message OperatorComponent {
+  // Definition name.
+  string definition_name = 1 [(google.api.field_behavior) = OPTIONAL];
+  // Operator definition.
+  OperatorDefinition definition = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Task.
+  string task = 3 [(google.api.field_behavior) = OPTIONAL];
+  // Input configuration of the component. JSON schema described in the operator definition.
+  google.protobuf.Struct input = 4 [(google.api.field_behavior) = OPTIONAL];
+  // Condition statement determining whether the component is executed or not.
+  string condition = 5 [(google.api.field_behavior) = OPTIONAL];
+}
+
+// IteratorComponent
+// Configures an iterator component for iteration.
```

Comment:
```suggestion
// Configures an iterator component. An iterator takes an array and executes an
// operation (defined by a set of nested components) on each of its elements.
// It can be regarded as triggering a sub-pipeline using the elements of an
// array as input.
```